src/server/ua_subscription_alarms_conditions.c

```diff
@@ -0,0 +1,2575 @@
+            CONDITION_ASSERT_RETURN_VOID(retval, "Set ConfirmedState/Id failed",);
+        }
+    }
+    else {
+        /* Set unconfirming time */
+        retval = UA_Server_writeObjectProperty_scalar(server, conditionNode, UA_QUALIFIEDNAME(0, CONDITION_FIELD_TIME),
+                                             &data->sourceTimestamp, &UA_TYPES[UA_TYPES_DATETIME]);
+        CONDITION_ASSERT_RETURN_VOID(retval, "Set deactivating Time failed", UA_NodeId_deleteMembers(&conditionNode););
+
+        /* Set ConfirmedState text to (Unconfirmed)*/
+        text = UA_LOCALIZEDTEXT(LOCALE, UNCONFIRMED_TEXT);
+        UA_Variant_setScalar(&value, &text, &UA_TYPES[UA_TYPES_LOCALIZEDTEXT]);
+        retval = UA_Server_setConditionField(server, conditionNode, &value, fieldConfirmedState);
+        UA_NodeId_deleteMembers(&conditionNode);
+        CONDITION_ASSERT_RETURN_VOID(retval, "Set Condition ConfirmedState failed",);
+    }
+}
+#endif//CONDITIONOPTIONALFIELDS_SUPPORT
+
+static void
+afterWriteCallbackActiveStateChange(UA_Server *server,
+                                    const UA_NodeId *sessionId, void *sessionContext,
+                                    const UA_NodeId *nodeId, void *nodeContext,
+                                    const UA_NumericRange *range, const UA_DataValue *data) {
+    UA_QualifiedName fieldEnabledState = UA_QUALIFIEDNAME(0, CONDITION_FIELD_ENABLEDSTATE);
+    UA_QualifiedName fieldActiveState = UA_QUALIFIEDNAME(0, CONDITION_FIELD_ACTIVESTATE);
+    UA_QualifiedName fieldActiveStateId = UA_QUALIFIEDNAME(0, CONDITION_FIELD_TWOSTATEVARIABLE_ID);
+    UA_LocalizedText text;
+    UA_Variant value;
+
+    UA_NodeId twoStateVariableNode;
+    UA_StatusCode retval = getFieldParentNodeId(server, nodeId, &twoStateVariableNode);
+    CONDITION_ASSERT_RETURN_VOID(retval, "No Parent TwoStateVariable found for given ActiveState/Id",);
+
+    UA_NodeId conditionNode;
+    retval = getFieldParentNodeId(server, &twoStateVariableNode, &conditionNode);
+    UA_NodeId_deleteMembers(&twoStateVariableNode);
+    CONDITION_ASSERT_RETURN_VOID(retval, "No Parent Condition found for given ActiveState",);
+
+    /* Get conditionSource */
+    UA_NodeId conditionSource;
+    retval = getNodeIdValueOfConditionField(server, &conditionNode, UA_QUALIFIEDNAME(0, CONDITION_FIELD_SOURCENODE), &conditionSource);
+    CONDITION_ASSERT_RETURN_VOID(retval, "ConditionSource not found", UA_NodeId_deleteMembers(&conditionNode););
+
+    UA_ActiveState lastActiveState = UA_INACTIVE, currentActiveState = UA_INACTIVE;
+    UA_Boolean isLimitalarm = false;
+
+    retval = getConditionActiveState(server, &conditionSource, &conditionNode, &lastActiveState, &currentActiveState, &isLimitalarm);
+    CONDITION_ASSERT_RETURN_VOID(retval, "ActiveState transition check failed", UA_NodeId_deleteMembers(&conditionNode);
+                                 UA_NodeId_deleteMembers(&conditionSource););
+
+    if(isLimitalarm == false) {
```

Comment:
Could you please explain the idea behind this code?

Reply:
The idea was to use it later with LimitAlarmType Conditions. At this point it will always evaluate to false to update the ActiveState.
In the Future implementation, if evaluates to true, we don't need to update the ActiveState, because it should be updated in another callback specific for the LimitAlarms.

Reply:
@sqadasi thank you for the explanation